### DISCUSSION TITLE
Mobile Phone (Samsung) Image Orientation wrong

### DISCUSSION THREAD

**Discussion by futureweb** [ImageMagick/ImageMagick#2597] [Help]
Hey there,
using "mogrify -auto-orient" on an Image Uploader for my Website. Now I noticed that Image is getting rotated upside down on some Images taken with different Samsung Mobile Phones.
Whether you turn the Phone to the right or to the left it's working or not (EXIF Orientation 1 vs 3)

Original Image
- http://in7.futureweb.at/temp/20200918_105759.jpg
- http://in7.futureweb.at/temp/20200918_105753.jpg

After running them through mogrify / resizing:
- https://www.futureweb.at.dev.futurecms.at/userdata/6348/gallery/15851/thumbnails/20200918_105759_2000x1329.jpg
- https://www.futureweb.at.dev.futurecms.at/userdata/6348/gallery/15851/thumbnails/20200918_105753_2000x1329.jpg

![grafik](https://user-images.githubusercontent.com/1698881/93582106-4ec87680-f9a2-11ea-8541-4b0ff7cc2d7e.png)

Already tried all sort of different fixes - but wasn't able to solve the Problem ... :-/
Hope someone here can give me a helping hand? ;-)

thx, bye from Austria
Andreas

**Comment by snibgo** [comment]
What version of IM, on what platform?

What was the exact command you used?

I tried:
```
magick mogrify -auto-orient x.jpg
```
... on the first image, using IM v7.0.8-64 and v7.0.10-27. Both versions gave the correct result.

**Reply by futureweb** [reply]
Already tried several different Versions of IM - all showed the same behaviour for me.
Also MagickStudio (https://archive.imagemagick.org/MagickStudio/scripts/MagickStudio.cgi) shows the same behaviour.
Of the 2 Pics linked in my Posting - 1 Pic is working fine, the other not ... depending if the Mobile Phone was turned to the left or to the right.

**Comment by snibgo** [comment]
With your second image, 20200918_105753.jpg, both IM and exiftool says it has normal orientation, aka "1", aka horizontal. So "-auto-orient" does not rotate it.

**Reply by futureweb** [reply]
This is one of those Files which are not rotated correctly:
http://in7.futureweb.at/temp/wrong.jpg
```
Orientation: BottomRight
exif:Orientation: 3
```

It's shown "upside down" within Image Metadata Viewer: http://exif.regex.info/exif.cgi and also if opened within Browser. But "BottomWrite" should be right I guess?! as it's rotated ...

After "/usr/bin/mogrify -auto-orient wrong.jpg" it show's:
http://in7.futureweb.at/temp/wrong_mogrify.jpg

```
Orientation: TopLeft
exif:Orientation: 1
```

Metadata Viewer and Browser still show the Image upside down?! But "TopLeft" is wrong - as it's still upside down?

Do I overlook here something?

**Comment by snibgo** [comment]
wrong.jpg contains bad metadata.

The image in wrong.jpg is the correct way up. It doesn't need any rotation.

But the metadata claims it needs to be rotated 180 degrees. This is wrong.

IM "-auto-rotate" will obey the false metadata, and will rotate the image by 180 degrees, so it is then inverted.

**Reply by futureweb** [reply]
Hey @snibgo - thx for that Info ... ! ;-)

What would be the best way to check which Orientation the JPG got?
I'm never sure if I see the original orientated Pic or the "exif orientation corrected" Pic ... ?

I was able to reproduce this behaviour with several different Samsung Mobiles, not sure if non-Samsung also write wrong metadata?

It depends if one tilts the Phone to the Right or to the Left when taking Landscape Photo ...
On one Side it always stores it as "Orientation: 1" ... the other Side it stores it as "Orientation: 3" ... but it seems the JPG Pixel Data is always orientated 'correct' ... (Orientation: 1)

Which is a pretty big Problem as all Users with Samsung Mobiles often get false rotated Images ... damn :-(

**Comment by snibgo** [comment]
You can get the metadata from exiftool or IM:
```
f:\web\im>exiftool wrong.jpg |grep rient
Orientation                     : Rotate 180

f:\web\im>%IMG7%magick wrong.jpg -verbose info: |grep rient
  Orientation: BottomRight
    exif:Orientation: 3
```

When I open wrong.jpg in Gimp v2.10.0, it says:
![wrongGimp](https://user-images.githubusercontent.com/33812368/94016061-92a5ec00-fda5-11ea-9d22-1bc83e58107e.jpg)

With human intelligence, I can see that the original is correct, and rotating the image would be wrong.

EDIT to add: On Samsung orientation problems, see also https://github.com/ImageMagick/ImageMagick/discussions/1959
